open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +255,450 @@ public static JSONArray appendDummyAccountID() {
         return accountsJSON;
 
     }
+
+        /**
+         * Method that consists the implementation for the validation of  payload and the consent,
+         * this method also invokes the relevant methods to populate data for each flow.
+         *
+         * @param consentResource Consent Resource parameter containing consent related information retrieved
+         *                        from database.
+         * @return ConsentDataJson array
+         */
+        public static JSONArray getConsentData(ConsentResource consentResource) throws ConsentException {
+
+            JSONArray consentDataJSON = new JSONArray();
+            try {
+
+                String receiptString = consentResource.getReceipt();
+                Object receiptJSON = new JSONParser(JSONParser.MODE_PERMISSIVE).parse(receiptString);
+
+                // Checking whether the request body is in JSON format
+                if (!(receiptJSON instanceof JSONObject)) {
+                    log.error(ErrorConstants.NOT_JSON_OBJECT_ERROR);
+                    throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                            ErrorConstants.NOT_JSON_OBJECT_ERROR);
+                }
+
+                if (!consentResource.getCurrentStatus().equals(ConsentExtensionConstants.AWAITING_AUTH_STATUS)) {
```

Comment:
Shall we change this to `ConsentExtensionConstants.AWAITING_AUTH_STATUS.equals(consentResource.getCurrentStatus())` to avoid NPE